File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.java
class Solution {
    public int countGoodSubstrings(String s) {
        int result = 0;

        //n-2개 -> asd, sdb, dbb, bbc,bca,cab,abc
        for(int i=0; i<s.length()-2; i++) {
            String tmpStr = s.substring(i, i+3);
            if((tmpStr.charAt(0) != tmpStr.charAt(1)) && (tmpStr.charAt(1) != tmpStr.charAt(2)) && (tmpStr.charAt(0) != tmpStr.charAt(2))){
                result++;
            }
        }

        return result;
    }
}

File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.kt
class Solution {
fun countGoodSubstrings(s: String): Int {
    var count = 0
    for (i in 0 until s.length -2) {
        val substring = s.substring(i, i + 3).map { it.toString() }
        val distinct = substring.distinct()
        if (distinct.size == 3) {
            count++
        }
    }
    return count
}
}